### PR TITLE
[Wallet] [RPC] Remove deprecated wallet rpc features from bitcoin_server

### PR DESCRIPTION
**Pull request by jnewbery**
The following rpc features were deprecated in V0.17:

- `validateaddress` returning wallet information about an address
- `signrawtransaction`

This PR fully removes those features. It can be merged once V0.17 has been branched from master.

### CODE REVIEW

**Comment by jnewbery** [comment]
open-closing this PR so the branch is available to everyone. This+ #10762 should allow server to be built without any dependencies on wallet.

Will re-open once V0.17 has been branched.

**Comment by dmoney0546** [comment]
Really.my thing i been looking for years

**Comment by jnewbery** [comment]
rebased

File: src/rpc/misc.cpp [hunk 1]
```diff
@@ -67,22 +62,14 @@ static UniValue validateaddress(const JSONRPCRequest& request)
...
-#ifdef ENABLE_WALLET
-        if (HasWallets() && IsDeprecatedRPCEnabled("validateaddress")) {
-            ret.pushKVs(getaddressinfo(request));
-        }
-#endif
-        if (ret["address"].isNull()) {
-            std::string currentAddress = EncodeDestination(dest);
-            ret.pushKV("address", currentAddress);
+        CScript scriptPubKey = GetScriptForDestination(dest);
+        ret.pushKV("scriptPubKey", HexStr(scriptPubKey.begin(), scriptPubKey.end()));;
```
Comment (MarcoFalke) [inline src/rpc/misc.cpp]: nit: No need for double semicolon?

**Review by maflcko — COMMENTED** [review]
utACK 1158febc93707910d783be863aa05c8f6cfedb2f

File: src/rpc/misc.cpp (hunk 1 above)
Comment (jnewbery) [reply src/rpc/misc.cpp]: thanks. Fixed

**Comment by jnewbery** [comment]
V0.17 has been branched. This is ready for review/merge

**Comment by jnewbery** [comment]
rebased

**Comment by MarcoFalke** [comment]
re-utACK f7e9e704683be4d6be4c9ea600a479bc02976313

**Comment by promag** [comment]
re-utACK f7e9e70.

**Comment by ptyx11** [comment]
What was the reasoning behind this? It seems unnecessary.

**Comment by achow101** [comment]
@ptyx11 This is part of the separation of the wallet from the node. Those RPCs were node RPCs which involved wallet things, so they needed to be removed. They have been deprecated for at least one major version with replacement RPCs already available.

**Pull request merged**